services/ingestor/src/main/kotlin/company/thewebhook/ingestor/Application.kt

```diff
@@ -47,16 +50,28 @@ fun Application.module() = launch {
             .lowercase()
             .split(",")
 
+    val serverHostRegexString =
+        environment.config.propertyOrNull("ingestor.validation.serverHostRegex")?.getString()
+            ?: throw ServerHostRegexException("Serverhost regex string not provided")
+
     val producer by inject<Producer<ByteArray>>()
     producer.connect(messageStoreConfig)
 
+    val serverHostRegex = Regex(serverHostRegexString)
+
     routing {
         HttpMethod.DefaultMethods.forEach {
             route("{...}", it) {
                 handle {
-                    val serverHost = call.request.origin.serverHost
+                    val serverHost = call.request.origin.serverHost.lowercase()
                     val remoteHost = call.request.origin.remoteHost
 
+                    if (!serverHostRegex.matches(serverHost)) {
+                        call.application.log.error("Invalid server host")
```

Comment:
We need a better log message. Also, include `serverHost` and `remoteHost` in the message.

Reply:
`Malformed Host header, serverHost and remoteHost must be valid hostnames`

How about this?

Reply:
Let's have it as this?
`Server host does not match the regex. Server Host: $serverHost, Remote Host: $remoteHost.`

Also, this needs to be a Bad Request, not Forbidden. Can you also send a message in the response? Maybe `HOST_INVALID` as a string should be fine.

Reply:
Okayy got it